addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -383,36 +381,34 @@ def test_redirect_to_records_nothread(self):
 
         # when being logged, cids should be based on current user's company unless
         # there is an access issue (not tested here, see 'test_redirect_to_records')
-        self.authenticate(self.user_admin.login, self.user_admin.login)
         for test_record in nothreads:
             for user_company in self.company_admin, self.company_2:
                 with self.subTest(record_name=test_record.name, user_company=user_company):
+                    self.authenticate(self.user_admin.login, self.user_admin.login)
```

Comment:
Why this change ? People won't log out log in each time they click on a link anyway ? Tests should still work as before, shouldn't they ?

Reply:
I think that, previous to this commit, the test didn't take into account the modification of the cookies when clicking on a link in a mail (done when loading the webclient based on the url cids). So in each of the subTests when reading from the cookies : https://github.com/odoo/odoo/blob/f71072de8414d84317e6eee2bb0ad2cae25b798c/addons/mail/controllers/mail.py#L74
The cookies are always the same, the ones of the authentication.

Now, as the modification of the cookies is done by the python code, in order to have the same results as before, and to avoid that one subTest have a non-desired effect in another subTest, we need to reset the cookies to the initial ones (after the authentication). So, I placed the authenticate process at the beginning of each subTest.

At least, is what I understood of the test.

Reply:
Then what happens if you click on multiple links in a given session ? That should probably be a new test that demonstrates the change linked to the new implementation ...